agent/engine/task_manager.go

```diff
@@ -252,6 +255,13 @@ func (mtask *managedTask) overseeTask() {
 	mtask.engine.wakeUpTaskQueueMonitor()
 	// TODO: make this idempotent on agent restart
 	go mtask.releaseIPInIPAM()
+
+	csiClient := csiclient.NewCSIClient(filepath.Join(csiclient.SocketHostPath, csiclient.ImageName, csiclient.SocketName))
```

Comment:
non-blocking but in future we should move the `csiClient` initialization into the `UnstageVolumes` method after checking that there is indeed an EBS volume attached to the task.